panel-app/src/components/store.ts

```diff
@@ -102,14 +124,7 @@ function updateState(newState): boolean {
 					return { id: getId(), type: item.type, path: item.path };
 				});
 
-			const formattedMetadata =
-				metadata &&
-				Object.keys(metadata).reduce((acc, key) => {
-					const item = metadata[key];
-					const { type, path, isBoolean } = item;
-					acc[getId()] = { name: key, type: type, path: path, ...(isBoolean && { isBoolean: isBoolean }) };
-					return acc;
-				}, {});
+			const formattedMetadata = metadata ? getFormattedMetadata('updateState', metadata) : {};
```

Comment:
You can move the check for `metadata` into `getFormattedMetadata` and have the function return an empty object `{}` when it's invalid.

Reply:
Thankyou